testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -79,10 +80,51 @@ class MetricFrameworkTests : public CppUnit::TestFixture
 
 protected:
 
+    void Test_valid_and_invalid_metric_names()
+    {
+        std::vector<std::string> invalid_names = {
+            {".startswithperiod"},
+            {"endswithperiod."},
+            {"9startswithnumber"},
+            {"has_underscore"},
+            {"has&^%$#@!-~`+=|[]{}?/<>,_specialcharacters"}
+        };
```

Comment:
I think "has.a.&" would currently pass?  One special character at the end of the string?